src/gc.c

```diff
@@ -1536,12 +1537,12 @@ static void gc_scan_obj_(jl_ptls_t ptls, jl_value_t *v, int d,
     }
     else {
         int nf = (int)jl_datatype_nfields(vt);
-        uint32_t npointers = vt->layout->npointers;
         nptr += (npointers & 0xff) << (npointers & 0x300);
-        for(int i=0; i < nf; i++) {
+        uint32_t offsets = ((uint32_t*)vt->layout)[-1];
```

Comment:
`jl_layout_offsets`? (I have no idea what this does, just basing it on the code here)